### ISSUE TITLE
Vjoy not detected as input device on Control Mapper when using Bridge

### ISSUE THREAD

**Issue by Bhgp-git** [SHWotever/SimHub#1755]
**Describe the bug**
VJoy input devices are recognized as controllers at simhub, but not recognized as input devices on control mapper plugin

**Simhub version**
9.5.4

**To Reproduce**
Add vjoy device
Try to add it on control mapper
Not available in control mapper but recognized as controller

**Expected behavior**
Map a virtual axis as clutch

**Additional context**
:et me know if there is anything else I can provide. I couldn't find a solution in the internet

**Comment by SHWotever** [comment]
Hi ! This is intentional, since The control mapper also uses vjoy as an output, only physical controllers (non vjoy) are allowed as an input to avoid creating a non sense endless loop (simhub feeding vjoy, mapped as an input, feeding vjoy etc ...)

**Comment by Bhgp-git** [comment]
Oh I see. I had to create a virtual axis using UCR at vjoy in order to use dual clutch paddles properly, since there is no way to merge axis using Simhub. Is there a way to bring in this feature as an "advanced mode", for example?

Thanks for the very fast response :)

**Comment by SHWotever** [comment]
Hi ! 

I've added multiple axis mapping for a single controller in the next version as well as response curves to manipulate the effective output range of each one : 

![image](https://github.com/user-attachments/assets/70e84db9-d9fd-455f-be5c-8e7b7192fad5)

**Comment by Bhgp-git** [comment]
> Hi !
> 
> I've added multiple axis mapping for a single controller in the next version as well as response curves to manipulate the effective output range of each one :
> 
> ![image](https://private-user-images.githubusercontent.com/2207331/393361827-70e84db9-d9fd-455f-be5c-8e7b7192fad5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MDk2OTQsIm5iZiI6MTczMzUwOTM5NCwicGF0aCI6Ii8yMjA3MzMxLzM5MzM2MTgyNy03MGU4NGRiOS1kOWZkLTQ1NWYtYmU1Yy04ZTdiNzE5MmZhZDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDZUMTgyMzE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjRhNDUxZDg5ZjJkN2U2ZjdkNWRiY2EzZTM2YTYzYjhlYTEzMDY1MmQxNDI4ZjQ3NDljOWRiNGNkNGJlNTg1NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.pFZmFnjE2wY-yU27grJmf_it1lZmva0B1wPgLUs_vd8)

Awesome! I will give it a try and come back here! From which version on is it available?

**Comment by Bhgp-git** [comment]
Hi, first of all many thanks for the implementation! It works almost perfectly as is.

I have found a few bugs:
- Right clicking or left clicking to add or remove nodes on the curve editor currently doesn't work.
- I would to like to split my clutch response perfectly 50/50 between the axis, but it doesn't allow me to start a curve with 50 response value. I had to split like this, which doesn't give me a perfectly linear response on the upper axis range (from 100 to 50)
- ![Screenshot 2024-12-11 145701](https://github.com/user-attachments/assets/e284f19b-047f-4287-be35-554bb9869a35)
- ![Screenshot 2024-12-11 145801](https://github.com/user-attachments/assets/4d052f97-c78a-4ef2-9d31-d5824648f89f)
- The lower end of the axis could be perfectly linear mapped :) (From 50 to 0)

It is plenty usable as is, but I would just like to point out what i found. Many thanks again for the implementation, I can get rid of vjoy now 👍

**Comment by SHWotever** [comment]
Hi ! Indeed the label is a little misleading I need to correct it.

Axis always need to start from 0 for obvious reasons ;) axis are taking the max, it's not cumulating.
Starting from 50 would make the finalk axis response start right from 50. For your purpose you would need more a "cumulating response" where both mapped axis would cumulate :
ie : left 0 to 50, right 0 to 50 : the sum is giving 0 to 100 when both are pressed. But in your example you are showing a third axis which I guess is acting independently as 0 100 ?

I've used 6 fixed points where first and last point can be moved (and intermediate points will be evenly spread) , adding and removing any random number of points was making it too complicated for the purpose.

**Comment by Bhgp-git** [comment]
> Hi ! Indeed the label is a little misleading I need to correct it.
> 
> Axis always need to start from 0 for obvious reasons ;) axis are taking the max, it's not cumulating. Starting from 50 would make the finalk axis response start right from 50. For your purpose you would need more a "cumulating response" where both mapped axis would cumulate : ie : left 0 to 50, right 0 to 50 : the sum is giving 0 to 100 when both are pressed. But in your example you are showing a third axis which I guess is acting independently as 0 100 ?
> 
> I've used 6 fixed points where first and last point can be moved (and intermediate points will be evenly spread) , adding and removing any random number of points was making it too complicated for the purpose.

Yes, exactly I would like to sum or average the response of both axis together, but it didn't work like this when I mapped both of them to 50, they were simply going both from 0 to 50. So I had to improvise and make one from 0 to 50 and the other one from 0-100, albeit very fastly from 0 to 50 and then linear from 60-100.

The third axis is my normal clutch pedal for other purposes, simply mapped from 0 to 100 linear :)
